Fix for Spring loaded throttle in manual throttle modes

jason4short opened the pull request:
This is a patch which introduces a throttle control for disengaging throttle when arming via mavlink while in a manual mode such as Acro or Stabilize. 

All other modes with alt hold don't need to check throttle position so this removes the "Arm: Throttle too high" check. 

UX : user enters stabilize mode, presses ARM on Droidplanner. motors spool up to idle and the user brings throttle to zero. Throttle engages at zero and as it is raised it works normally.

R-Lefebvre commented:
Jason, I love sprung-center-throttle as well.  But it won't become the norm anytime soon. There is a huge installed base of users who have legacy RCTx hardware with unsprung throttles.  Most will not run out and buy a new Tx no matter awesome it is.  So we need to protect for that logic still.

R-Lefebvre commented:
IMO, sprung-center-throttle goes hand in hand with a Ch8 Throttle_Kill function.  That's how I'm doing it, but the logic is in my RCTx firmware.  Basically, if Ch8 is low, throttle is off.  When Ch8 is high, the motors do not stop, ever, even if the stick is on the bottom.  This solves many of the current problems with SCT arming.  It doesn't matter where the stick is, if Ch8 is not low, you can't arm.  So you can arm with the throttle anywhere.  However, the motors will not start until the Ch8 is set high by a switch.  Once that happens, throttle goes to the stick control point.

While flying, if you bump the bottom of the stick, the motors won't stop.  This is really good for Stab and Acro.  Conversely, if you ever want to absolutely kill the motors, switch Ch8 low.

This method of having an absolute kill switch on Ch8 has been used for a long long time on RC Helis, and it works well.  I think it would be a better way of dealing with SCT.

jason4short commented:
This small patch only affects the spring loaded throttles, and can be tweaked not to affect any other typical system. 
Jason


> On Dec 31, 2014, at 4:07 PM, Robert Lefebvre <notifications@github.com> wrote:
> 
> Jason, I love sprung-center-throttle as well. But it won't become the norm anytime soon. There is a huge installed base of users who have legacy RCTx hardware with unsprung throttles. Most will not run out and buy a new Tx no matter awesome it is. So we need to protect for that logic still.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/diydrones/ardupilot/pull/1716#issuecomment-68476000>.
>

jason4short commented:
OK, added the check for disarming, and returned throttle too high message for now for non-manual modes. Also added Drift mode to manual modes since the throttle is basically manual.
Just a note,but manual_flight_mode is ambiguous. Is it manual throttle or no GPS assist? I think it's only used to track manual throttle.

lucasdemarchi commented:
@rmackay9 @jason4short  any news for this one? I've seen some RCs with spring-loaded and it would be nice to support them, otherwise I don't have on myself.

R-Lefebvre commented:
I'm pretty sure this, or something like it, has gone into master.  Solo has a spring loaded Throttle.

lucasdemarchi commented:
@R-Lefebvre thanks. Adding @jschall here then. @jschall is this something on solo's repository only or do we already have everything in master?

davidbuzz commented:
This PR is very old, we think it's probably no longer relevant so we are closing it now, please let us know if this is in error
